packages/block-library/src/image/editor.scss

```diff
@@ -22,6 +22,10 @@ figure.wp-block-image:not(.wp-block) {
 		margin-top: -9px;
 		margin-left: -9px;
 	}
+
+	&:not(.is-style-rounded) > div {
```

Comment:
Can this be serialized to the inner element (`img`) in the edit function by using `__experimentalSkipSerialization`? Adding JS/CSS/etc. to a block so it can use some block supports is a smell that we need to do something differently. 

This is now relevant for this PR, in which we're making the block selector to target the inner `img`. See https://github.com/WordPress/gutenberg/pull/34073/files#r688580510

Reply:
I believe it can be serialized to the `img` element now. This wasn't possible when this PR was done as the `__experimentalSkipSerialization` option didn't exist yet.

There is actually already a [PR](https://github.com/WordPress/gutenberg/pull/31366) that moves the border radius support styles to the inner image element as part of adopting border color and width support for the image block. https://github.com/WordPress/gutenberg/pull/31366

I can sort out the style mentioned here as part of that.